worker/storage/provisioner.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package storage
+
+import (
+	"path/filepath"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"launchpad.net/tomb"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/api/environment"
+	"github.com/juju/juju/api/provisioner"
+	"github.com/juju/juju/environs/config"
+	corestorage "github.com/juju/juju/storage"
+	"github.com/juju/juju/storage/provider"
+	"github.com/juju/juju/worker"
+)
+
+var logger = loggo.GetLogger("juju.storage.provisioner")
+
+var _ Provisioner = (*storageProvisioner)(nil)
+
+// TODO - add tests
```

Comment:
Not sure there's any point, because this all needs to change pretty significantly.